### PR TITLE
chore: Add check to ensure token value is string and add test

### PR DESCRIPTION
**Pull request by rubencarvalho**
### Description

In trying to construct a `PropertyFilterOperatorExtended<string[]>` in a PropertyFilter, a token value type of `string[]` doesn’t actually work, but not until you try to edit. At token edit time, there’s a util that tries to manipulate the value as a string that fails (trying to call `toLowerCase()` on it). This wasn't caught by TypeScript because we don't have an ESLint rule for checking unsafe calls of `any` typed values.

This PR makes the handling logic in the util more robust by checking the type first. 

Related links, issue #, if available: AWSUI-21107

### How has this been tested?

<details>
   <summary>Review checklist</summary>

_The following items are to be evaluated by the author(s) and the reviewer(s)._

#### Correctness

- _Changes include appropriate documentation updates._
- _Changes are backward-compatible if not indicated, see [`CONTRIBUTING.md`](https://github.com/cloudscape-design/components/blob/main/CONTRIBUTING.md#public-apis)._
- _Changes do not include unsupported browser features, see [`CONTRIBUTING.md`](https://github.com/cloudscape-design/components/blob/main/CONTRIBUTING.md#browsers-support)._
- _Changes were manually tested for accessibility, see [accessibility guidelines](https://cloudscape.design/foundation/core-principles/accessibility/)._

#### Security

- _If the code handles URLs: all URLs are validated through [the `checkSafeUrl` function](https://github.com/cloudscape-design/components/blob/main/src/internal/utils/check-safe-url.ts)._

#### Testing

- _Changes are covered with new/existing unit tests?_
- _Changes are covered with new/existing integration tests?_
</details>

---

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/property-filter/utils.ts [hunk 1]
```diff
@@ -69,7 +69,10 @@ export function matchTokenValue(token: Token, filteringOptions: readonly Interna
       // exact match found: return it
       return { ...token, value: option.value };
     }
-    if (token.value.toLowerCase() === (option.label ?? option.value ?? '').toLowerCase()) {
+    if (
+      typeof token.value === 'string' &&
```
Comment (timogasda) [inline src/property-filter/utils.ts:73]: If we have to make this type check, doesn't that mean that our Typescript types are inaccurate? Can we fix that?

Comment (rubencarvalho) [reply src/property-filter/utils.ts:73]: There is an ESLint rule that can detect this.

It currently shows:
25 reports of such “[unsafe calls](https://typescript-eslint.io/rules/no-unsafe-call/)”
61 reports of “[unsafe assignments](https://typescript-eslint.io/rules/no-unsafe-assignment/)” to variables
43 reports of “[unsafe arguments](https://typescript-eslint.io/rules/no-unsafe-argument/)” to functions
and 14 reports of “[unsafe returns](https://typescript-eslint.io/rules/no-unsafe-return)” in functions

We have this task as a internal BRF ticket.

Comment (rubencarvalho) [reply src/property-filter/utils.ts:76]: So... looking at the type definition for the `token.value`, coming from the [collection hooks package](https://github.com/cloudscape-design/collection-hooks/blob/main/src/interfaces.ts#L144), it has to be `any`: by default, the token value is a `string` - however, when a custom property is used, the token value can be `any`.

I added a comment here to also make it more clear.

Comment (timogasda) [reply src/property-filter/utils.ts:76]: Got it, thank you!

**Pull request merged**